pkg/controller/tokenrequestor/reconciler.go

```
@@ -238,6 +240,16 @@ func (r *Reconciler) computeRequeueAfterDuration(ctx context.Context, secret *co
 		}
 	}
 
+	// Check if the token belongs to the current ServiceAccount by verifying the UID (it could have been deleted and
+	// recreated with the same name which results in a change of the UID, i.e., the old token is now invalid).
+	// If there is a mismatch, a new token needs to be requested.
+	serviceAccountTokenUID, err := kubernetes.ExtractServiceAccountUID(string(secret.Data[resourcesv1alpha1.DataKeyToken]))
+	if err != nil {
+		log.V(1).Info("Secret token did not contain valid service account reference", "error", err)
```

Nit: You could add a comment why the error is not returned.

Nit 2: Why not `log.Error()`?